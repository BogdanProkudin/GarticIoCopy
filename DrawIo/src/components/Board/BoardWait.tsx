import Lottie from "react-lottie";
import { useAppDispatch, useAppSelector } from "../../store/hook";
import styles from "./styles.module.scss";
import LottieSettings from "../../tools/Animation - 1711477541401.json";
import { socket } from "../../socket";
import { setIsGameStarted, setIsRoundEnd } from "../../store/slices/roomInfo";
import { memo, useEffect } from "react";
import { handleGameStartButton } from "../../utils/handleStartGame";
import { setIsUserDraw } from "../../store/slices/userInfo";
import { useGetRoomIdFromUrl } from "../../hooks/useGetRoomIdFromUrl";

export const useGameSelectors = () => {
  return {
    activeIndex: useAppSelector((state) => state.drawThema.activeIndex),
    isGameStarted: useAppSelector((state) => state.drawThema.isGameStarted),
    roomUsers: useAppSelector((state) => state.drawThema.roomUsers),
    host: useAppSelector((state) => state.drawThema.host),
    selectedThema: useAppSelector((state) => state.drawThema.selectedThema),
  };
};

const BoardWait = () => {
  const dispatch = useAppDispatch();
  const roomId = useGetRoomIdFromUrl();
  const { activeIndex, roomUsers, host, selectedThema } = useGameSelectors();
  const userNameStorage = localStorage.getItem("userName");

  const maxGamePoints = useAppSelector(
    (state) => state.drawThema.maxRoomPoints
  );
  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: LottieSettings,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice",
    },
  };
  if (!roomUsers) {
    return <div>IN BOARD WAIT ZERO USERS</div>;
  }

  useEffect(() => {
    const onGameStarted = () => {
      dispatch(setIsUserDraw(true));
      dispatch(setIsRoundEnd(true));

      dispatch(setIsGameStarted(true));
    };

    socket.on("gameStarted", onGameStarted);

    return () => {
      socket.off("gameStarted", onGameStarted);
    };
  }, [dispatch]);

  const waitingTextStyles = {
    top:
      roomUsers.length > 1 && host.hostName !== userNameStorage ? "3rem" : "",
  };

  const animationContainerStyles = {
    top:
      roomUsers.length > 1 && host.hostName !== userNameStorage
        ? "5rem"
        : "2rem",
  };

  const waitingForOwnerTextStyles = {
    top:
      roomUsers.length > 1 && host.hostName !== userNameStorage
        ? "21.5rem"
        : "",
  };

  return (
    <div className={styles.board_wait_container}>
      <h1 className={styles.game_room_wait_text} style={waitingTextStyles}>
        WAITING
      </h1>
      <div
        className={styles.board_wait_animation}
        style={animationContainerStyles}
      >
        <Lottie
          options={defaultOptions}
          height={190}
          width={280}
          style={{ marginTop: "3.5rem" }}
          isClickToPauseDisabled
        />
      </div>
      {roomUsers.length <= 1 && (
        <span className={styles.game_room_wait_small_text}>
          Waiting for players
        </span>
      )}
      {roomUsers.length > 1 && host.hostName === userNameStorage && (
        <>
          <span className={styles.game_room_wait_small_text}>
            Players are waiting for you to start
          </span>
          <button
            data-testid={"button-id"}
            onClick={() =>
              handleGameStartButton({
                words: selectedThema.words,
                roomUsers,
                maxGamePoints,
                roomId,
                activeIndex,
                dispatch: dispatch,
              })
            }
            className={styles.game_room_start_button}
          >
            <div />
            <strong>PLAY!</strong>
          </button>
        </>
      )}
      {roomUsers.length > 1 && host.hostName !== userNameStorage && (
        <span
          className={styles.game_room_wait_small_text}
          style={waitingForOwnerTextStyles}
        >
          Waiting for the owner to start
        </span>
      )}
    </div>
  );
};

export default memo(BoardWait);
