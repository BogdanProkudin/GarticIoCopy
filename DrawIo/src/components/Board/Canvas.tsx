import { fabric } from "fabric";
import {
  useState,
  useRef,
  useEffect,
  useCallback,
  MutableRefObject,
  memo,
} from "react";
import { socket } from "../../socket";
import { useAppDispatch, useAppSelector } from "../../store/hook";
import { setDrawColor } from "../../store/slices/drawInfo";
import styles from "./styles.module.scss";
import { useInitializeCanvas } from "../../hooks/useInitializeCanvas";
import { useMouseEvents } from "../../hooks/useMouseEvents";
import { useUpdateDrawingParams } from "../../hooks/useUpdateDrawingParams ";

interface DrawingCanvasProps {
  roomId: string;
  contextRef: MutableRefObject<CanvasRenderingContext2D | null>;
  drawRef: any;
}

const DrawingCanvas: React.FC<DrawingCanvasProps> = ({
  roomId,
  contextRef,
  drawRef,
}) => {
  const brushWidth = useAppSelector((state) => state.drawInfo.brushWidth);
  const activeUser = useAppSelector((state) => state.drawThema.activeUser);
  const drawColor: string = useAppSelector((state) => state.drawInfo.drawColor);
  const userNameStorage = localStorage.getItem("userName");
  const activeTool: string = useAppSelector(
    (state) => state.drawInfo.activeTool
  );
  const canvasRef = useRef<HTMLCanvasElement | null>(null);

  // useEffect(() => {
  //   const canvas = new fabric.Canvas(canvasRef.current, {
  //     width: 650,
  //     height: 400,
  //     backgroundColor: "white",
  //     isDrawingMode: false,
  //     selectionFullyContained: false,
  //     selection: false,
  //   });

  //   canvas.hoverCursor = "crosshair";
  //   canvas.moveCursor = "crosshair";
  //   canvas.defaultCursor = "crosshair";

  //   drawRef.current = canvas;
  //   contextRef.current = canvas.getContext() as CanvasRenderingContext2D;

  //   socket.on("getDraw2", (data: any) => {
  //     if (
  //       data.type === "path" &&
  //       drawRef.current &&
  //       activeUser.userName &&
  //       activeUser.userName !== userNameStorage
  //     ) {
  //       const path = new fabric.Path(data.path, data.options);
  //       drawRef.current.add(path);
  //       drawRef.current.renderAll();
  //       canvas.selection = false;
  //       drawRef.current.forEachObject((obj: any) => {
  //         obj.selectable = false;
  //       });
  //     }
  //   });

  //   canvas.on("path:created", (e: any) => {
  //     const path = e.path;
  //     const pathData = {
  //       type: "path",
  //       roomId,
  //       path: path.path,
  //       options: path.toObject([
  //         "left",
  //         "top",
  //         "fill",
  //         "stroke",
  //         "strokeWidth",
  //       ]),
  //     };
  //     canvas.renderAll();
  //     socket.emit("drawing2", pathData);
  //   });

  //   return () => {
  //     canvas.dispose();
  //   };
  // }, [roomId, activeUser.userName, userNameStorage]);

  useInitializeCanvas({
    userNameStorage,
    activeUser,
    roomId,
    canvasRef,
    contextRef,
    drawRef,
  });
  useUpdateDrawingParams(
    drawRef,
    activeUser,
    userNameStorage!,
    activeTool,
    drawColor,
    brushWidth
  );

  const { handleMouseDown, handleMouseMove, handleMouseUp } = useMouseEvents(
    drawRef,
    activeUser,
    userNameStorage!,
    activeTool,
    drawColor,
    roomId
  );
  useEffect(() => {
    const canvas = drawRef.current;
    if (!canvas) return;

    canvas.on("mouse:down", handleMouseDown);
    canvas.on("mouse:up", handleMouseUp);

    if (activeTool === "pen" || activeTool === "eraser") {
      canvas.on("mouse:move", handleMouseMove);
      return () => canvas.off("mouse:move", handleMouseMove);
    }

    return () => {
      canvas.off("mouse:down", handleMouseDown);
      canvas.off("mouse:up", handleMouseUp);
    };
  }, [handleMouseDown, handleMouseUp, handleMouseMove, activeTool]);

  return (
    <div>
      <canvas className={styles.canvas_content} ref={canvasRef} />
    </div>
  );
};

export default memo(DrawingCanvas);
