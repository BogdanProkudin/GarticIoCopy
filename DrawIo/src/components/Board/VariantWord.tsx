import styles from "./styles.module.scss";
type VariantWordProps = {
  word: string;
};
const VariantWord: React.FC<VariantWordProps> = ({ word }) => {
  return (
    <>
      <span className={styles.suggestion_word}>{word}</span>
      <button>
        <div />
        <strong>Draw</strong>
      </button>
    </>
  );
};
export default VariantWord;
