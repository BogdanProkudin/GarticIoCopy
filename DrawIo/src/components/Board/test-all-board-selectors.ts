import { initialStateProps } from "../../store/slices/roomInfo";

interface IdrawThema {
  drawThema: initialStateProps;
}

const boardWaitstate = {
  drawThema: {
    roomUsers: [
      { id: "1", userName: "owner", isActive: false },
      { id: "2", userName: "owner2", isActive: false },
    ],
    host: "owner",
    activeIndex: 0,
    selectedThema: {
      name: "Животные",
      icon: "",
      words: ["собака", "крокодил"],
    },
  },
};

const boardState = {
  drawThema: {
    selectedThema: { name: "", icon: "", words: [] },
    selectedPoints: 70,
    selectedPlayers: 5,
    roomData: {
      id: "",
      host: "",
      players: 0,
      points: 0,
      thema: {
        icon: "",
        name: "",
        words: [],
      },
      usersInfo: [],
    },
    roomUsers: [],
    host: "",
    activeUser: {
      host: "",
      userName: "",
      userId: "",
      userAvatar: "",
      isActive: false,
      userPoints: 0,
    },
    maxRoomPoints: 0,
    roundCount: 0,
    activeIndex: 0,
    isGameStarted: false,
    choosedWord: "",
    isUserDraw: false,
    isRoundEnd: false,
    isUserWonGame: false,
    isAllUsersGuessed: false,
    isOneUserGuessed: false,
    chosenWords: [],
    activeTool: "",
    drawColor: "black",
    brushWidth: "5",
    toolsPanel: false,
    winners: [
      {
        isActive: false,
        userAvatar: "",
        userName: "",
        userPoints: 0,
      },
    ],
  },
};

export const testUseBoardWaitSelector = (
  f: (arg0: {
    drawThema: {
      roomUsers: { id: string; userName: string; isActive: boolean }[];
      host: string;
      activeIndex: number;
      selectedThema: { name: string; icon: string; words: string[] };
    };
  }) => any
) => f(boardWaitstate);

export const testUseBoardSelector = (f: IdrawThema | any) => {
  return f(boardState);
};
