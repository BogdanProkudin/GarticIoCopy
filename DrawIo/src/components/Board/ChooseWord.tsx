import { useEffect, memo } from "react";

import LottieSettings from "../../tools/Animation - 1711653829862.json";
import Lottie from "react-lottie";
import styles from "./styles.module.scss";
import { socket } from "../../socket";
import {
  setChoosedWord,
  setChoosedWordsList,
  setChosenWords,
  setIsGameStarted,
  setIsRoundTimerOver,
  setRoundCount,
} from "../../store/slices/roomInfo";

import { useAppDispatch, useAppSelector } from "../../store/hook";

import ProgressBar from "./Timers/IntervalTimer";
import VariantWord from "./VariantWord";
import WaitTurn from "./WaitTurn";
import BoardInactive from "./roundResult/BoardInactive";
import { setIsUserDraw } from "../../store/slices/userInfo";
import { setToolsPanel } from "../../store/slices/drawInfo";
import axios from "axios";
import { useGetRoomIdFromUrl } from "../../hooks/useGetRoomIdFromUrl";

const ChooseWord = memo(function ChooseWord() {
  const roomId = useGetRoomIdFromUrl();

  const { activeUser, choosedWordsList, chosenWords, isRoundEnd } =
    useAppSelector((state) => ({
      activeUser: state.drawThema.activeUser,
      chosenWords: state.drawThema.chosenWords,
      isRoundEnd: state.drawThema.isRoundEnd,
      choosedWordsList: state.drawThema.choosedWordsList,
    }));
  const userName = localStorage.getItem("userName");
  const dispatch = useAppDispatch();
  const handleChooseDrawWord = (choosedWord: string) => {
    const sendWordChoosed = async (url: string) => {
      try {
        const response = await axios.post(url, {
          headers: {
            "Content-Type": "application/json",
          },

          roomId,
        });
        if (response.data) {
          socket.emit("wordChoosed", { choosedWord, roomId });
        }
      } catch (error) {
        console.error("Error:", error);
      }
    };
    const url = "http://localhost:3000/wordChoosed";

    sendWordChoosed(url);
  };
  const defaultOptions = {
    loop: true,
    autoplay: true,

    animationData: LottieSettings,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice",
    },
  };

  // useEffect(() => {
  //   const getWhoIsNext = async () => {
  //     const foundActiveUser = await roomUsers.find(
  //       (el: { userName: any }) => el.userName === activeUser.userName
  //     );
  //     if (foundActiveUser) {
  //       const activeUserIndex = await roomUsers.indexOf(foundActiveUser);

  //       if (userName === roomUsers[activeUserIndex].userName) {
  //         setTimeout(() => {}, 1000);
  //       }
  //     }
  //   };
  //   getWhoIsNext();
  // }, [roomUsers]);

  useEffect(() => {
    socket.on("getWordChoosed", (data: any) => {
      dispatch(setChoosedWord(data));
      dispatch(setToolsPanel(true));
      dispatch(setRoundCount());
      dispatch(setChoosedWordsList([...choosedWordsList, data]));
      dispatch(setIsUserDraw(false));

      dispatch(setIsGameStarted(true));
    });
    socket.on("gameWords", async (words) => {
      dispatch(setChosenWords(words));
    });

    return () => {
      socket.off("gameWords");
      socket.off("getWord");
    };
  }, []);

  return (
    <div className={styles.container}>
      <div className={styles.white_background}>
        {activeUser.userName === userName && isRoundEnd ? (
          <div className={styles.your_turn_draw_container}>
            <h1>Its your Turn</h1>
            <span>Choose a word to Draw</span>
            <div className={styles.choose_word_animation}>
              <Lottie options={defaultOptions} />
            </div>
            <h2 className={styles.or_text}>or</h2>
            <div className={styles.board_choose_word_container} />
            <div className={styles.suggestion_variants_container}>
              {chosenWords.map((word) => {
                return (
                  <div
                    key={word}
                    onClick={() => handleChooseDrawWord(word)}
                    className={styles.suggestion_word_container}
                  >
                    <VariantWord word={word} />
                  </div>
                );
              })}
            </div>
            <div style={{ marginTop: "0.6rem" }}>
              <ProgressBar />
            </div>
          </div>
        ) : isRoundEnd ? (
          <WaitTurn />
        ) : (
          <BoardInactive />
        )}
      </div>
    </div>
  );
});

export default ChooseWord;
