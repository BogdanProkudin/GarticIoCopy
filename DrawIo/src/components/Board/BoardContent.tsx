import Lottie from "react-lottie";
import BoardWait from "./BoardWait";
import ChooseWord from "./ChooseWord";
import WordGuessSuccess from "./roundResult/WordGuessSuccess";
import styles from "./styles.module.scss";
interface Props {
  isGuessedAnimationFinished: boolean;
  setIsGuessedAnimationFinished: Dispatch<SetStateAction<boolean>>;
}
import LottieSettings from "../../tools/Animation - 1712170538790.json";
import { Dispatch, SetStateAction, useEffect } from "react";
import { useAppSelector } from "../../store/hook";
import UsersNotGuessed from "./roundResult/UsersNotGuessed";

const BoardContent: React.FC<Props> = ({
  isGuessedAnimationFinished,
  setIsGuessedAnimationFinished,
}) => {
  const {
    choosedWord,
    isOneUserGuessed,
    isUsersNotGuessed,
    isGameStarted,
    isAllUsersGuessed,
  } = useAppSelector((state) => ({
    choosedWord: state.drawThema.choosedWord,
    isOneUserGuessed: state.userInfo.isOneUserGuessed,
    isUsersNotGuessed: state.userInfo.isUsersNotGuessed,
    isGameStarted: state.drawThema.isGameStarted,
    isAllUsersGuessed: state.userInfo.isAllUsersGuessed,
  }));
  const defaultLottieOptions = {
    loop: false,
    autoplay: true,

    animationData: LottieSettings,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice",
    },
  };

  const handleAnimationComplete = () => {
    setIsGuessedAnimationFinished(true);
  };

  if (!isGameStarted) {
    return <BoardWait />;
  }

  if (choosedWord.length < 1) {
    return <ChooseWord />;
  }

  if (isUsersNotGuessed) {
    return <UsersNotGuessed />;
  }

  if (isAllUsersGuessed) {
    return <WordGuessSuccess />;
  }

  if (
    !isAllUsersGuessed &&
    isOneUserGuessed &&
    !isGuessedAnimationFinished &&
    isUsersNotGuessed
  ) {
    return (
      <div className={styles.one_user_guessed_animation_container}>
        <div className={styles.one_user_guessed_animation}>
          <Lottie
            options={defaultLottieOptions}
            isClickToPauseDisabled
            eventListeners={[
              {
                eventName: "complete",
                callback: handleAnimationComplete,
              },
            ]}
          />
        </div>
      </div>
    );
  }

  if (isAllUsersGuessed) {
    return <WordGuessSuccess />;
  }

  return <></>;
};

export default BoardContent;
