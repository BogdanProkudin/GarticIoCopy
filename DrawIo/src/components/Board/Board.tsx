import React, { useEffect, useState, useCallback, memo } from "react";
import { socket } from "../../socket";
import styles from "./styles.module.scss";
import { useAppDispatch, useAppSelector } from "../../store/hook";
import { useNavigate } from "react-router-dom";
import BoardContent from "./BoardContent";
import useMousePosition from "./DrawSettings/ToolsPicker/EraserPosition";
import { handleNextUserCall } from "../../hooks/getActiveUser";
import RoundTimer from "./Timers/RoundTimer";
import { logDOM } from "@testing-library/react";
import { useDrawing } from "../../hooks/useDrawing";
import axios from "axios";
import { useGetRoomIdFromUrl } from "../../hooks/useGetRoomIdFromUrl";

type BoardProps = {
  roomId: string;
  contextRef: React.RefObject<any>;
  drawRef: React.RefObject<any>;
};

// interface User {
//   id: string;
//   userName: string;
//   isActive: boolean;
//   userPoints: number;
// }

// interface UserData {
//   users: User[];
// }

const Board: React.FC<BoardProps> = ({ contextRef, drawRef }) => {
  const userNameLocalStorage = localStorage.getItem("userName");
  const [isGuessedAnimationFinished, setIsGuessedAnimationFinished] =
    useState(false);
  const mousePosition = useMousePosition();
  const activeUser = useAppSelector((state) => state.drawThema.activeUser);
  const dispatch = useAppDispatch();
  const roundCount = useAppSelector((state) => state.drawThema.roundCount);
  const activeTool = useAppSelector((state) => state.drawInfo.activeTool);
  const roomUsers = useAppSelector((state) => state.drawThema.roomUsers);
  const roomId = useGetRoomIdFromUrl();
  const userNameStorage = localStorage.getItem("userName");
  const isAllUsersGuessed = useAppSelector(
    (state) => state.userInfo.isAllUsersGuessed
  );

  const toolsPanel = useAppSelector((state) => state.drawInfo.toolsPanel);
  const isGameStarted = useAppSelector(
    (state) => state.drawThema.isGameStarted
  );
  const maxGamePoints = useAppSelector(
    (state) => state.drawThema.maxRoomPoints
  );

  useEffect(() => {
    console.log("slibidirizz");

    socket.on("getNextUserCall", (data) => {
      handleNextUserCall({
        dispatch,
        roundCount,
        setIsGuessedAnimationFinished,
        roomUsers,
        maxGamePoints,
        data,
      });
    });

    return () => {
      socket.off("getNextUserCall", handleNextUserCall);
    };
  }, []);

  const { handleDrawing } = useDrawing({
    activeUser,
    drawRef,
    contextRef,
    userNameLocalStorage,
  });

  useEffect(() => {
    socket.on("getDraw", handleDrawing);
    return () => {
      socket.off("getDraw", handleDrawing);
    };
  }, [handleDrawing]);

  return (
    <>
      {activeTool === "eraser" && (
        <div
          className={styles.eraser_square}
          style={{
            left: mousePosition.x - 25,
            top: mousePosition.y + 40,
          }}
        ></div>
      )}

      <BoardContent
        isGuessedAnimationFinished={isGuessedAnimationFinished}
        setIsGuessedAnimationFinished={setIsGuessedAnimationFinished}
      />
      {isGameStarted && toolsPanel && !isAllUsersGuessed && <RoundTimer />}
    </>
  );
};

export default memo(Board);
