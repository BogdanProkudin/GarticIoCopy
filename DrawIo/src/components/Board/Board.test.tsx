import { render } from "@testing-library/react";
import Board from "./Board";
import { socket } from "../../socket";
import { handleNextUserCall } from "../../hooks/getActiveUser";

import { useAppSelector, useAppDispatch } from "../../store/hook";

jest.mock("react-redux");
jest.mock("../../socket", () => ({
  socket: {
    on: jest.fn(),
    off: jest.fn(),
  },
}));
jest.mock("../../hooks/getActiveUser", () => ({
  handleNextUserCall: jest.fn(),
}));
const setIsGuessedAnimationFinished = jest.fn();

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn((initial) => [initial, setIsGuessedAnimationFinished]),
}));
jest.mock("../../store/slices/drawThema", () => ({
  setIsUserWonGame: jest.fn(),
  setWinners: jest.fn(),
  setRoundCount: jest.fn(),
}));

const state = {
  drawThema: {
    activeUser: { userName: "testUser" },
    roundCount: 0,
    activeTool: "pen",
    roomUsers: [
      { userName: "owner", isActive: false, userPoints: 0 },
      { userName: "user2", isActive: false, userPoints: 0 },
      { userName: "user3", isActive: false, userPoints: 0 },
    ],
    maxRoomPoints: 100,
  },
};

describe("Компонент Board", () => {
  const dispatch = jest.fn();
  const mockUseAppSelector = useAppSelector as jest.MockedFunction<
    typeof useAppSelector
  >;
  const mockUseAppDispatch = useAppDispatch as jest.MockedFunction<
    typeof useAppDispatch
  >;

  beforeEach(() => {
    jest.clearAllMocks();
    mockUseAppDispatch.mockImplementation(() => dispatch);
  });

  test("handleNextUserCall вызывается при событии getNextUserCall от сокета", () => {
    const initialData = {
      users: [
        { userName: "owner", isActive: false, userPoints: 0 },
        { userName: "user2", isActive: false, userPoints: 0 },
        { userName: "user3", isActive: false, userPoints: 0 },
      ],
    };
    mockUseAppSelector.mockImplementation((f: any) => f(state));
    const socketOnSpy = jest.spyOn(socket, "on");

    render(
      <Board
        contextRef={{ current: "" }}
        drawRef={{ current: "" }}
        roomId={"test"}
      />
    );

    // Симуляция события getNextUserCall
    const onCall = socketOnSpy.mock.calls.find(
      (call) => call[0] === "getNextUserCall"
    );
    if (onCall) {
      const eventHandler = onCall[1];
      eventHandler(initialData);
    }

    // Проверка вызова handleNextUserCall
    expect(handleNextUserCall).toHaveBeenCalled();
    expect(handleNextUserCall).toHaveBeenCalledWith({
      data: initialData,
      setIsGuessedAnimationFinished,
      dispatch,
      roomUsers: state.drawThema.roomUsers,

      roundCount: state.drawThema.roundCount,
      maxGamePoints: state.drawThema.maxRoomPoints,
    });

    // Проверка вызова dispatch
  });
});
