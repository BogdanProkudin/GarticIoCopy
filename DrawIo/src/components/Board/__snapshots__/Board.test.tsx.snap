// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render board with 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <h1>
            Success
          </h1>
          <div>
            <div
              aria-label="animation"
              role="button"
              style="width: 100%; height: 100%; overflow: hidden; margin: 0px auto; outline: none;"
              tabindex="0"
              title=""
            />
          </div>
          <span>
            every body guessed correct
          </span>
          <div
            style="top: 14rem;"
          >
            <div
              style="width: 100%; height: 10px; background-color: rgb(5, 187, 242);"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <h1>
          Success
        </h1>
        <div>
          <div
            aria-label="animation"
            role="button"
            style="width: 100%; height: 100%; overflow: hidden; margin: 0px auto; outline: none;"
            tabindex="0"
            title=""
          />
        </div>
        <span>
          every body guessed correct
        </span>
        <div
          style="top: 14rem;"
        >
          <div
            style="width: 100%; height: 10px; background-color: rgb(5, 187, 242);"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
